### DISCUSSION TITLE
Difficulties in requesting gRPC using access token

### DISCUSSION THREAD

**Discussion by Shortood** [cloudforet-io/community#146] [Q&A]
The extraction of access token and refresh token was successful

How should I put in the access token when I make a request with gRPC?

 I don't know how to deal with "injected from auth (required)".
 
When you add access token to the bearer token of the gRPC request, you get the error "ERROR_REQUIRED_PARAMETER: Required parameter. (key = domain_id)"

![image](https://github.com/user-attachments/assets/02940ea9-9ef7-44b5-8d9d-d4bd3009d507)

![image](https://github.com/user-attachments/assets/41b744fc-b710-402d-85c9-1700de5a7ec9)

**Comment by junho100** [comment]
I think required metadata is not set in the request.
When you make request, you have to set "x_domain_id" in metadata.
Check out the codes in python core.

https://github.com/cloudforet-io/python-core/blob/239631e27c9f5a3d07061fc12c3721283cfe3d73/src/spaceone/core/handler/authentication_handler.py#L160

https://github.com/cloudforet-io/python-core/blob/239631e27c9f5a3d07061fc12c3721283cfe3d73/src/spaceone/core/service/__init__.py#L46

https://github.com/cloudforet-io/python-core/blob/239631e27c9f5a3d07061fc12c3721283cfe3d73/src/spaceone/core/error.py#L72

**Comment by ImMin5** [comment]
Hi, @Shortood 

Thank you for opening the discussion with us.

Currently, Cloudforet 2.0 is not stable version, so we recommend using version 1.12. 
Please note that `x_domain_id` does not work in version 1.12.

It seems that the issue you mentioned is related to Cloudforet 2.0. To resolve it, please issue an App key from the Console.

![스크린샷 2024-07-18 오후 12 37 27](https://github.com/user-attachments/assets/5c21bd86-524b-4a14-84c8-9b1f7da13514)

![스크린샷 2024-07-18 오후 12 39 51](https://github.com/user-attachments/assets/7ab355bf-6d90-48dc-aeb3-69435b94474f)
